test-suite/js/src/test/scala/org/scalajs/testsuite/jsinterop/NestedJSClassTest.scala

```diff
@@ -628,6 +656,29 @@ object NestedJSClassTest {
     class InnerScalaClass(val zzz: Int)
   }
 
+  class DoublyNestedInnerObject_Issue4114 extends js.Object {
+    object middle extends js.Object {
+      object inner extends js.Object {
+        var x = 1
+      }
+    }
+  }
+
+  object TriplyNestedObject_Issue4114 extends js.Object {
+    object middle extends js.Object {
+      object inner extends js.Object {
+        var x = 1
+      }
+
+      class innerClass extends js.Object {
```

Comment:
Naming convention nitpick:
```suggestion
      class InnerClass extends js.Object {
```